tangelo/algorithms/classical/fci_solver.py

```diff
@@ -131,3 +145,147 @@ def get_rdm(self):
+
+        Returns:
+            numpy.array: One-particle RDM.
+            numpy.array: Two-particle RDM.
+
+        Raises:
+            RuntimeError: If method "simulate" hasn't been run.
+        """
+
+        # Check if Full CI has been performed
+        if self.ciwfn is None:
+            raise RuntimeError("FCISolver: Cannot retrieve RDM. Please run the 'simulate' method first")
+        if self.degenerate_mo_energies:
+            warnings.warn("Degenerate orbitals are present in the molecule. The fci calculation is performed "
+                          "without symmetry so the rdms may not correspond to the integrals in molecule. A c1 "
+                          "version of the molecule with the correct integrals is present as FCISolverPsi4.molecule.")
+        one_rdm = np.asarray(self.ciwfn.get_opdm(0, 0, "SUM", False))
+        two_rdm = np.asarray(self.ciwfn.get_tpdm("SUM", True))
+
+        return one_rdm, two_rdm
+
+
+fci_solver_dict = {"pyscf": FCISolverPySCF, 'psi4': FCISolverPsi4}
+
+
+def get_fci_solver(molecule: SecondQuantizedMolecule, solver: Union[None, str, Type[ElectronicStructureSolver]] = default_fci_solver, **kwargs):
+    """Return requested target FCISolver object.
+
+    Args:
+        molecule (SecondQuantizedMolecule) : Molecule
+        solver (string or Type[ElectronicStructureSolver] or None): Supported string identifiers can be found in
+            fci_solver_dict (from tangelo.algorithms.classical.fci_solver). Can also provide a user-defined backend (child to ElectronicStructureSolver class)
+        kwargs: Other arguments that could be passed to a target. Examples are solver type etc.
+     """
+
+    if solver is None:
+        if isinstance(molecule.solver, IntegralSolverPySCF):
+            solver = FCISolverPySCF
+        elif isinstance(molecule.solver, IntegralSolverPsi4):
+            solver = FCISolverPsi4
+        elif default_fci_solver is not None:
+            solver = default_fci_solver
+        else:
+            raise ModuleNotFoundError(f"One of the backends for {list(fci_solver_dict.keys())} needs to be installed to use FCISolver"
+                                      "without providing a user-defined implementation.")
+
+    # If target is a string use target_dict to return built-in backend
+    elif isinstance(solver, str):
+        try:
+            solver = fci_solver_dict[solver.lower()]
+        except KeyError:
+            raise ValueError(f"Error: backend {solver} not supported. Available built-in options: {list(fci_solver_dict.keys())}")
+    elif not issubclass(solver, ElectronicStructureSolver):
+        raise TypeError(f"Target must be a str or a subclass of ElectronicStructureSolver but received class {type(solver).__name__}")
+
+    return solver(molecule, **kwargs)
+
+
+def FCISolver(molecule: SecondQuantizedMolecule, solver: Union[None, str, Type[ElectronicStructureSolver]] = default_fci_solver, **kwargs):
```

Comment:
2 comments:

- I think your code blob is literally equivalent (but less efficient) that writing `FCISolver = get_fci_solver`, since they have the exact same behaviour. You are defining an alias, except you duplicated some code and you have an indirection in the call.

- What's the point, here ? Apart from PEP8 naming convention, why isn't it perfectly suitable for people to use `FCISolver`, which is how all our tests and solvers work ?

We moved from `Simulator` to `get_backend` because we had to move from a legacy name (Backend was more appropriate) and we kept the former around a while to ensure user code would not immediately break (we should totally remove it in the next PR btw).
https://github.com/goodchemistryco/Tangelo/blob/main/tangelo/linq/simulator.py

Here, FCISolver seems to be still perfectly relevant and desirable to me, I don't really see why we have `get_fci_solver`.

Reply:
- Not according to my development environment (VSCode). It labels FCISolver as a variable when using `FCISolver = get_fci_solver`, although when you mouse over it, it does return the docstring of a function
<img width="949" alt="Screenshot 2023-06-15 at 10 11 01 AM" src="https://github.com/goodchemistryco/Tangelo/assets/84878946/dc3e1ef3-e941-4155-a492-00dd97e3285e">

-  With the separate function, it labels it as a function, which is correct. There is nothing wrong with it outside of it violating PEP8 so it looks a little strange.
<img width="960" alt="Screenshot 2023-06-15 at 10 11 37 AM" src="https://github.com/goodchemistryco/Tangelo/assets/84878946/019ec49f-dbce-4b64-a4be-482763a35b4e">

In general, I prefer `get_fci_solver` as this is similar to `get_backend`. We could have just as easily changed `Simulator` to `Backend` but chose `get_backend`.

If we decide that `FCISolver` is the way to go then I will remove `get_fci_solver`.

Reply:
Right now I kept both. But now `FCISolver` is a class that has a solver attribute which is `solver` that takes on either `FCISolverPySCF` or `FCISolverPsi4` or user defined. `get_fci_solver` returns `FCISolverPySCF` or `FCISolverPsi4`. We can change the names but I like having both. 

If users want to access attributes without using `fcisolver.solver.whatever`. They can create a solver with `get_fci_solver`.

Where possible, I default to using `IntegralSolverName` but sometimes this is not possible. Especially with the upcoming CCSD solver which ignores frozen orbitals if you try to feed it a reference wavefunction is psi4.